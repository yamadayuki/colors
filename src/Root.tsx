import { useEffect, useState } from "react";
import { useLocation } from "react-use";
import { router } from "./routes";

export const Root = () => {
  const location = useLocation();
  const [Component, setComponent] = useState<JSX.Element | null>(null);

  useEffect(() => {
    router
      .resolve(location.pathname || "")
      .then(component => setComponent(component));
  }, [router, location.pathname]);

  return Component;
};
