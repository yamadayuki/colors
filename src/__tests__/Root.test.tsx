import React from "react";
import { render } from "react-testing-library";
import { Root } from "../Root";

it("should render null in first rendering", () => {
  const { container } = render(<Root />);

  expect(container.firstChild).toMatchSnapshot();
});
