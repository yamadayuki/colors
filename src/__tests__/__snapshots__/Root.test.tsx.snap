// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render null in first rendering 1`] = `null`;
